docs: add API routes page (#528)
tuono-labs/tuono: apps/documentation/src/routes/documentation/routing/api-routes.mdx

import MetaTags from '@/components/MetaTags'

<MetaTags
  title="Tuono - API routes"
  canonical="https://tuono.dev/documentation/routing/api-routes"
/>

import Breadcrumbs from '@/components/Breadcrumbs'

<Breadcrumbs breadcrumbs={[{ label: 'API routes' }]} />

# API routes

## Overview

> API routes provide a solution to build a public API with Tuono.

Reviewer: ```suggestion
API routes offer a solution for building a public API using Tuono.
```

> Any file inside the folder `src/routes/api` is mapped to `/api/*`
> path and will be treated as an API endpoint instead of a React route.

Reviewer: ```suggestion
Any file inside the `src/routes/api` folder is mapped to the `/api/*` path 
and will be treated as an API endpoint, rather than a React route.
```

> Tuono server is internally managed by
> [axum](https://docs.rs/axum/latest/axum/); the API system allows creating APIs
> following most of the axum syntax. Tuono re-exports axum `tuono_lib::axum::*` to
> be seamlessly used within any Tuono app.

Reviewer: ```suggestion
Tuono server is internally managed by [axum](https://docs.rs/axum/latest/axum/). 
The API system allows you to create APIs using most of the axum syntax.
Tuono re-exports axum `tuono_lib::axum::*` to be seamlessly used within any Tuono app.
```

The major differences are:

> - The function to be matched with Tuono file system routing needs to have the
>   `tuono_lib::api(method)` macro attribute on top of it

Reviewer: ```suggestion
- The function to be matched with Tuono's file system routing must have the
  `tuono_lib::api(method)` macro attribute on top of it.
```

> - The first function argument alway is `tuono_lib::Request`

Reviewer: ```suggestion
- The first argument of the function is always a `tuono_lib::Request`.
```

> Tuono API supports [dynamic routes](/documentation/routing/dynamic-routes).

Reviewer: ```suggestion
Tuono also supports [dynamic routes](/documentation/routing/dynamic-routes) for APIs.
```

## Example

> If you create `src/routes/api/books.rs` with the following file content any HTTP
> GET request to `/api/books` will return the list of queried books as JSON.

Reviewer: ```suggestion
If you create `src/routes/api/books.rs` with the following content, 
any HTTP GET request to `/api/books` will return a list of the queried books as JSON.
```

```rs
use tuono_lib::Request;
use tuono_lib::axum::response::IntoResponse;

use serde_json::{Value, json};

#[tuono_lib::api(GET)]
async fn book_list(req: Request) -> Json<Value> {
  let books = get_books_from_db().await;

  Json(json!(books))
}
```

> > Unlike React routes API must not return `tuono_lib::Response`

Reviewer: ```suggestion
> Unlike React routes, API must not return `tuono_lib::Response`
```

> The same file can contain more HTTP methods

Reviewer: ```suggestion
The same file can contain multiple HTTP methods.
```
